include/threads.h

```diff
@@ -26,9 +26,24 @@
 #ifndef _THREADS_H_
 #define _THREADS_H_
 
-void logging_thread_cleanup(void *arg);
+#include <pthread.h>
+#include <unistd.h>
 
 void *logging_thread(void *arg);
 void *reading_thread(void *arg);
 
+// see https://blog.memzero.de/pthread-cancel-noexcept/
```

Comment:
great to have those macros!

i would suggest to provide some inline documentation on how those are to be used here, instead of only the external reference, just two or three lines... i.e.:
"vzlogger uses `pthread_cancel()` to stop threads, which is not safe for C++ code that might invoke destructors,
 this macro is to be placed around any code that your meter spends significant amounts of time in,
 but which may not contain C++ code that might destroy objects., see .... for details."

Reply:
done